src/go/cmd/http-relay-server/server/broker.go

```diff
@@ -259,9 +263,17 @@ func (r *broker) SendResponse(resp *pb.HttpResponse) error {
 
 	// Writing to this channel will notify consumers which are waiting for data
 	// on the channel returned by RelayRequest().
-	pr.responseStream <- resp
-
+	select {
+	case pr.responseStream <- resp:
+		// This is limited by how fast the user client consumes the stream so we should not
+		// lock here.
```

Comment:
Can we try to rewrite the comment?
- what is "this"?
- we still hold the lock, but you added a way to cancel, right?